### PR TITLE
Add documentation on Bluetooth Mesh and Bluetooth LE. Minor other additions.

### PR DESCRIPTION
**Pull request by mrquincle**

### CODE REVIEW

**Review by ArrowAcrobatics — APPROVED** [review]

File: docs/MESH_PROTOCOL.md [hunk 1]
```diff
@@ -156,7 +166,7 @@ uint8_t | List size | 1 | Size of tracked devices list.
 
 Type | Name | Length | Description
 --- | --- | --- | ---
-uint8_t | Stone ID | 1 | ID of stone that requests for a sync.
+stone_id_t | Stone ID | 1 | ID of stone that requests for a sync.
```
Comment (vliedel) [inline docs/MESH_PROTOCOL.md:169]: I'd prefer not to have typedefs in the protocol doc.

**Review by vliedel — CHANGES_REQUESTED** [review]
Nice work!
After resolving my 1 comment, it can be merged.

File: docs/MESH_PROTOCOL.md (hunk 1 above)
Comment (mrquincle) [reply docs/MESH_PROTOCOL.md:169]: Ah, I just made it consistent with the rest. :-)

Comment (mrquincle) [reply docs/MESH_PROTOCOL.md]: Okay, `s/stone_id_t/uint8_t/g` and at the same time added `_t` postfix where this was forgotten.

Comment (vliedel) [reply docs/MESH_PROTOCOL.md]: Ah, right, the mesh protocol doc isn't on par with the protocol doc: it's copy pasted structs basically.

Comment (vliedel) [reply docs/MESH_PROTOCOL.md]: We don't use the _t prefix in the protocol doc though, but well, i'm fine with this.

**Pull request merged**